src/ucp/wireup/wireup.c

```diff
@@ -783,6 +783,10 @@ static ucs_status_t ucp_wireup_msg_handler(void *arg, void *data,
 
     UCS_ASYNC_BLOCK(&worker->async);
 
+    if (worker->am_handlers_removed) {
+        goto out;
+    }
```

Comment:
how can we reach this code if we removed handlers?

Reply:
w/o this synchronization put in place, ucp_wireup_msg_handler() might process message after ucp_worker_remove_am_handlers() called within ucp_worker_destroy(), we may finally get [this warning](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34944&view=logs&j=c02db851-0d7a-5132-2265-ebd0aa7c6d37&t=a86ba9bf-30f1-543e-254c-1e7feba6151c&l=31763)

Reply:
we got the root cause for [this warning](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34944&view=logs&j=c02db851-0d7a-5132-2265-ebd0aa7c6d37&t=a86ba9bf-30f1-543e-254c-1e7feba6151c&l=31763). w/ the PR, `ucp_worker_discard_uct_ep_cleanup() `could trigger `ucp_worker_iface_deactivate() `which restores `iface->am[UCP_AM_ID_WIREUP]` removed by `ucp_worker_remove_am_handlers() `eariler. i think it should be ok to reorder these steps a bit to resolve [this warning](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34944&view=logs&j=c02db851-0d7a-5132-2265-ebd0aa7c6d37&t=a86ba9bf-30f1-543e-254c-1e7feba6151c&l=31763).